python_modules/dagster/dagster/_core/definitions/asset_sensor_definition.py

```diff
@@ -65,6 +65,8 @@ class AssetSensorDefinition(SensorDefinition):
             object to target with this sensor.
         jobs (Optional[Sequence[Union[GraphDefinition, JobDefinition, UnresolvedAssetJobDefinition]]]):
             (experimental) A list of jobs to be executed when the sensor fires.
+        tags (Optional[Mapping[str, object]]): A set of key-value tags that annotate the sensor and can
+            be used for searching and filtering in the UI. Values that are not already strings will be serialized as JSON.
```

Comment:
For asset definition tags, we only allow string values. Any downside to doing that here as well?

I think the main value of JSON tags is for use cases beyond searching and filtering in the UI, which we don't want to support.

Also tiny nit: line length

Reply:
k8s integration unfortunately relies on tags as json values. We would need a plan there I think

Reply:
But these aren't passed to the k8s integration, right? If they're just definition tags.

Reply:
Also note that the current `tags` param on `ScheduleDefinition` is annotated `Mapping[str, str]`

Reply:
Should have phrased this more exactly. I falsey assumed we were relying on tags at the job-level, but I think it is just an op-level. So thankfully I think we can do what schedule does. 👍🏻

Reply:
If we normalize everything to strings immediately, why disallow stuff that is JSON-serializable since we allow it in some places? Seems more user-friendly to keep our tags APIs consistent?

Reply:
I think the downside of allowing `object` is it makes it seem like JSON-style objects are a reasonable tag value, while I think we want to communicate that it isn't.  I think it's unfortunate that we need to maintain backwards compatibility with this pattern in some cases, but think there's long term cost to allowing it to spread more broadly.

I think the general rule would be that that parameters corresponding to definition tags don't accept JSON-style objects.

Reply:
ok, I've updated the PR to require str values